multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ExtractBatchedServicesWithResolvedDynamicParametersStep.java

```diff
@@ -109,11 +116,25 @@ private boolean isServiceInstanceGuidRequired(Set<DynamicResolvableParameter> dy
     private void setServicesToCreate(ProcessContext context, List<CloudServiceInstanceExtended> servicesCalculatedForDeployment) {
 
         List<CloudServiceInstanceExtended> servicesToCreate = servicesCalculatedForDeployment.stream()
-                                                                                             .filter(CloudServiceInstanceExtended::isManaged)
+                                                                                             .filter(
+                                                                                                 CloudServiceInstanceExtended::isManaged)
                                                                                              .collect(Collectors.toList());
         getStepLogger().debug(Messages.SERVICES_TO_CREATE, SecureSerialization.toJson(servicesToCreate));
         context.setVariable(Variables.SERVICES_TO_CREATE, servicesToCreate);
         context.setVariable(Variables.SERVICES_TO_CREATE_COUNT, servicesToCreate.size());
     }
 
+    private void checkForDuplicatedServiceNameFields(List<CloudServiceInstanceExtended> resolvedServiceInstances) {
```

Comment:
this is only in the 'batchedServices...' step - is there an alternative flow in the diagram that doesn't split the service in batches and will it not validate in that case?